File: AI/ml/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/SVR_MASTER.py
# SUPPORT VECTOR REGRESSION TEMPLATE

##################################
### check code     : cmd + i   ###
### run code       : cmd enter ###
### set working dir: Fn F5     ###
##################################

# Import libraries

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values # A MATRIX
y = dataset.iloc[:, 2].values # A VECTOR

# Splitting the dataset into the Training set and Test set
# from sklearn.cross_validation import train_test_split
# X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

# Feature Scaling (optional) necessary if there are Euclidian distance to be processed
# Necessary here as SVR class doesn't do it 
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = sc_y.fit_transform(y)

# Fitting SVR to the dataset 
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf')
regressor.fit(X, y)

# Predicting a new result with Polynominal Regression
y_pred = sc_y.inverse_transform(regressor.predict(sc_X.transform(np.array([6.5]))))

# Visialize the SVR results (CEO is considered outlier :( )
plt.scatter(X, y, color='red')  # actual
plt.plot(X, regressor.predict(X), color='blue')  # predicted
plt.title('Truth or Bluff (SVR MODEL)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

File: AI/ml/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/SVR_MASTER.R
############### R ####################
####  Data Preprocessing Template ####
############### R ####################
######## Contains all modules ########
############### R ####################
### check code      : Fn F1        ###
### run code        : cmd enter    ###
### set working dir : files 'more' ###
######################################


# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Splitting the dataset into the Training and Test set >>> Lets not do that here, 10 records only
#install.packages('caTools')
#library(caTools)
#set.seed(123)
#split = sample.split(dataset$Purchased, SplitRatio = 0.8)
#training_set = subset(dataset, split == TRUE)
#test_set = subset(dataset, split == FALSE)

# Feature Scaling (optional) necessary if there are Euclidian distance to be processed
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

# Fitting SVR to the datase
install.packages('e1071')
library(e1071)
regressor = svm()


# Single Predicting new result
y_pred = predict(regressor, data.frame(Level = 6.5))
summary(y_pred)

# Visualize SVR Model
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  # actual values
  geom_point(aes(x = dataset$Level , y = dataset$Salary),
             colour = "red") +
  # prediction
  geom_line(aes(x = dataset$Level , y = predict(regressor, newdata = dataset)), 
            colour = "blue") +
  ggtitle('Truth or Bluff (SVR MODEL)') +
  xlab('Level') +
  ylab('Salary')